prog/download_boot_image.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class Prog::DownloadBootImage < Prog::Base
+  subject_is :sshable, :vm_host
+
+  def image_name
+    @image_name ||= frame.fetch("image_name")
+  end
+
+  def custom_url
+    @custom_url ||= frame.fetch("custom_url")
+  end
+
+  label def start
+    vm_host.update(allocation_state: "draining")
+    hop_wait_draining
+  end
+
+  label def wait_draining
+    unless vm_host.vms.empty?
+      nap 15
+    end
+    sshable.cmd("sudo rm -f /var/storage/images/#{image_name.shellescape}.raw")
+    hop_download
+  end
+
+  label def download
+    case sshable.cmd("common/bin/daemonizer --check download_#{image_name.shellescape}")
+    when "Succeeded"
+      hop_activate_host
+    when "NotStarted"
+      sshable.cmd("common/bin/daemonizer 'host/bin/download-boot-image #{image_name.shellescape} #{custom_url.shellescape}' download_#{image_name.shellescape}")
+    when "Failed"
+      puts "#{vm_host} Failed to download '#{image_name}' image"
+    end
+
+    nap 15
+  end
+
+  label def activate_host
```

Comment:
I think we should re-learn the storage after downloading image.

Reply:
Should we add relearn to `Prog::DownloadBootImage`, or should we add the relearn label to the vm host nexus and trigger it with a semaphore?

Reply:
Budding `Prog::LearnStorage` from `Prog::DownloadBootImage` seems like most simple option.